sdk/trts/trts_veh.cpp

```diff
@@ -722,6 +740,42 @@ extern "C" sgx_status_t trts_handle_exception(void *tcs, outside_exitinfo_t *u_o
     return SGX_ERROR_ENCLAVE_CRASHED;
 }
 
+// For now, Occlum user space can have at most two inconsistent ranges.
+// The first range must exist
+#define OCCLUM_USER_SPACE_RANGE_NUM 2
+static sgx_addr_range_t g_occlum_user_space[OCCLUM_USER_SPACE_RANGE_NUM] = {{0,0}, {0,0}};
+
+static bool is_address_in_occlum_user_space(uintptr_t addr)
+{
+    sgx_addr_range_t range_a = g_occlum_user_space[0];
+    sgx_addr_range_t range_b = g_occlum_user_space[1];
+    assert(range_a.start != 0);
+    return (addr >= range_a.start && addr < range_a.end) || (range_b.start != 0 && addr >= range_b.start && addr < range_b.start);
+}
+
+static bool is_occlum_user_space_exception(sgx_exception_info_t *info)
+{
+    // Occlum doesn't register the exception handler
+    if (g_occlum_user_space_exception_handler == NULL || g_occlum_user_space[0].start == 0) {
+        return false;
+    }
+
+    uintptr_t faulting_addr = info->exinfo.faulting_address;
+    uintptr_t rip = info->cpu_context.REG(ip);
+    // Either faulting address or RIP points to the Occlum's userspace
+    // For #UD exception (rdtsc, cpuid, etc), the faulting address is zero, we need to check rip
+    // For Occlum kernel triggered #PF, the rip is not in user space, we need to check the faulting address
+    return is_address_in_occlum_user_space(faulting_addr) || is_address_in_occlum_user_space(rip);
+}
+
+void sgx_register_exception_handler_for_occlum_user_space(sgx_addr_range_t *user_space_ranges, sgx_exception_handler_t exception_handler)
+{
+    std::copy(user_space_ranges, user_space_ranges + OCCLUM_USER_SPACE_RANGE_NUM, g_occlum_user_space);
```

Comment:
Use `memcpy` instead of `std::copy`.  The C++ std should not be imported in trts library.